### DISCUSSION TITLE
Cronjob not running the latest version of notify

### DISCUSSION THREAD

**Discussion by njmulsqb** [projectdiscovery/notify#85] [Q&A] [answered]
Hi,

I was using notify version 0.0.2 on my ubuntu machine, when I realized that this version does not have -bulk flag support so I again ran the go install command that is given on readme file to install the latest version.

Now the issue is, when I run notify using my terminal myself it shows that the version is 1.0.0 and the bulk command is working fine but when I run the same command through cronjob it throws the error.

I checked the version ran by the cronjob by running notify -version through cron and it returned version 0.0.2.

Somehow, cron isnt able to fetch the latest notify version.

Why is it so and whats the solution?

**Comment by forgedhallpass** [comment]
Could you please show a screenshot/paste the command and the error you are receiving?

**Comment by njmulsqb** [comment]
> Could you please show a screenshot/paste the command and the error you are receiving?

![image](https://user-images.githubusercontent.com/47940898/133606236-ea5ec5d7-3520-41e4-bdfd-8a86d5c38c02.png)

**Comment by njmulsqb** [comment]
`notify -version` is showing 1.0.0 whereas `sudo notify -version` is using 0.0.2

**Comment by forgedhallpass** [comment]
Could you also do a `which notify` and `sudo which notify` and/or a `locate notify`? You might have two versions of `notify` installed on your machine. If you are using `go get ...` to install, it usually puts under `$GOPATH/` but if the environmental variable is not set, it might download it under `$HOME/go` as a default, hence running the command with `sudo` could result in a different `$HOME` directory compared to your user home.

**Comment by njmulsqb** [comment]
> Could you also do a `which notify` and `sudo which notify` and/or a `locate notify`? You might have two versions of `notify` installed on your machine. If you are using `go get ...` to install, it usually puts under `$GOPATH/` but if the environmental variable is not set, it might download it under `$HOME/go` as a default, hence running the command with `sudo` could result in a different `$HOME` directory compared to your user home.

root@ubuntu-s-1vcpu: which notify
/root/go/bin/notify
root@ubuntu-s-1vcpu:  sudo which notify
/usr/local/bin/notify

**Comment by forgedhallpass** [comment]
Now you need to check whether those are symbolic links and if they point to the same binary or not. I'd assume no, which means that you have two versions installed on your machine.

**Comment by njmulsqb** [comment]
> Now you need to check whether those are symbolic links and if they point to the same binary or not. I'd assume no, which means that you have two versions installed on your machine.

so how can I now get rid of this issue and just have one copy on my machine that is the latest one?

**Comment by forgedhallpass** [comment]
You can run `/root/go/bin/notify -version` and then `/usr/local/bin/notify -version` to see which one is the old version.
You can use `ls -l /usr/local/bin/notify` to see where does it point. If that is the old version then you can delete it, together with where it points.

You can also try `GO111MODULE=on go get -u -v github.com/projectdiscovery/notify/cmd/notify`, both with and without `sudo`, or `go clean -i github.com/projectdiscovery/notify/cmd/notify...` and reinstall again with the command above.

**Comment by njmulsqb** [comment]
> You can run `/root/go/bin/notify -version` and then `/usr/local/bin/notify -version` to see which one is the old version.
> You can use `ls -l /usr/local/bin/notify` to see where does it point. If that is the old version then you can delete it, together with where it points.
> 
> You can also try `GO111MODULE=on go get -u -v github.com/projectdiscovery/notify/cmd/notify`, both with and without `sudo`, or `go clean -i github.com/projectdiscovery/notify/cmd/notify...` and reinstall again with the command above.

Hi, I have removed both the installation with `rm -rf` now after again installing notify, `sudo notify` is saying that command is not found.

**Comment by forgedhallpass** [comment]
You should be able to find it under `$HOME/go/bin/notify` or `$GOPATH/bin/notify`. If you want it to be usable from anywhere, you'll have to create a symbolic link `ln -s <the_path_to_notify> /usr/local/bin/notify`.

**Reply by njmulsqb** [reply]
I am tired of different configurations but dont know somehow still notify is causing issues
I am running this command
`sudo /root/go/bin/notify -data /root/automation/results/semrush/subdomains/outOfScope.txt -bulk`
and its working perfectly, but when I run the exact same through cron, it fails with following error
![image](https://user-images.githubusercontent.com/47940898/135019225-db480d41-f476-4055-af4b-6269e76ab284.png)

`sudo which notify` gives command not found thats why I am running the binary with absolute path now. Whats wrong?

**Reply by parrasajad** [reply, marked as answer]
@njmulsqb this issue (#80 ) is fixed in dev and will be released soon, for now you can install dev version by running `go install github.com/projectdiscovery/notify/cmd/notify@dev`
